skyportal/tests/frontend/test_source_list.py

```diff
@@ -1,119 +1,183 @@
+    status, data = api(
+        'POST',
+        'sources',
+        data={
+            'id': f'{obj_id}',
+            'ra': 234.22,
+            'dec': -22.33,
+            'redshift': 0.153,
+            'altdata': {'simbad': {'class': 'RRLyr'}},
+            'transient': False,
+            'ra_dis': 2.3,
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token_two_groups,
+    )
+    assert status == 200
+    assert data['data']['id'] == f'{obj_id}'
+
+    # filter on the object id
+    obj_button = driver.wait_for_xpath("//input[@name='sourceID']")
+    obj_button.clear()
+    obj_button.send_keys(obj_id)
     submit = "//button[contains(.,'Submit')]"
     driver.click_xpath(submit)
-    driver.wait_for_xpath_to_be_unclickable(next_button)
 
+    # find the name of the newly added source
+    driver.wait_for_xpath(f"//a[contains(@href, '/source/{obj_id}')]")
 
-@pytest.mark.flaky(reruns=2)
-def test_jump_to_page_invalid_values(driver, user, public_group, upload_data_token):
-    obj_id = str(uuid.uuid4())
+    # find the date it was saved
+    saved_at_element = driver.wait_for_xpath(
+        f"//*[text()[contains(., '{t1.strftime('%Y-%m-%dT%H:%M')}')]]"
+    )
+    saved_group1 = parser.parse(saved_at_element.text + " UTC")
+    assert abs(saved_group1 - t1) < timedelta(seconds=2)
+
+    # check the redshift shows up
+    driver.wait_for_xpath(f"//*[text()[contains(., '{'0.153'}')]]")
+
+    # little triangle you push to expand the table
+    driver.click_xpath("//*[@id='expandable-button']")
+
+    # make sure the div containing the individual source appears
+    driver.wait_for_xpath(f'//tr[@data-testid="groupSourceExpand_{obj_id}"]')
+
+    driver.wait_for_xpath("//*[@class='vega-embed']")
+
+    # post a taxonomy and classification
+    status, data = api(
+        'POST',
+        'taxonomy',
+        data={
+            'name': "test taxonomy" + str(uuid.uuid4()),
+            'hierarchy': taxonomy,
+            'group_ids': [public_group.id, public_group2.id],
+            'provenance': f"tdtax_{__version__}",
+            'version': __version__,
+            'isLatest': True,
+        },
+        token=taxonomy_token_two_groups,
+    )
+    assert status == 200
+    taxonomy_id = data['data']['taxonomy_id']
+
+    status, data = api(
+        'POST',
+        'classification',
+        data={
+            'obj_id': obj_id,
+            'classification': 'Algol',
+            'taxonomy_id': taxonomy_id,
+            'probability': 1.0,
+            'group_ids': [public_group.id],
+        },
+        token=classification_token_two_groups,
+    )
+    assert status == 200
+
+    # check the classification shows up (it should not show up without a page refresh!)
+    try:
+        driver.wait_for_xpath(f"//*[text()[contains(., '{'Algol'}')]]", timeout=1)
+
+    except TimeoutException:
+        pass  # the classification should not appear on its own, so we ignore this error
```

Comment:
```suggestion
    # check the classification doesn't shows up (it should not show up without a page refresh!)
    driver.wait_for_xpath_to_disappear(f"//*[text()[contains(., '{'Algol'}')]]", timeout=1)
```
Sorry @guynir42 one more thing. When you test for things to not be present, you should use the `wait_for_xpath_to_disappear` instead of the `try-catch` pattern I think - for readability, performance, and consistency.

Reply:
Thanks! I didn't know about that one...